aiohttp/connector.py

```diff
@@ -1085,6 +1085,11 @@ async def _create_direct_connection(
 
         host = req.url.raw_host
         assert host is not None
+        # Replace multiple trailing dots with a single one.
+        # A trailing dot is only present for fully-qualified domain names.
+        # See https://github.com/aio-libs/aiohttp/pull/7364.
+        if host.endswith("."):
```

Comment:
This makes sense if there's at least two dots:
```suggestion
        if host.endswith(".."):
```